corehq/apps/app_manager/static/app_manager/js/visit_scheduler.js

```diff
@@ -323,7 +323,7 @@ hqDefine('app_manager/js/visit_scheduler', function () {
 
             self.relevancy = scheduleRelevancy.wrap(data.schedule);
             var xmlRe = /\s+|<+|>+|&+|"+|'+/g;
-            self.schedule_form_id = ko.observable(data.schedule_form_id).snakeCase(xmlRe);
+            self.schedule_form_id = ko.observable(data.schedule_form_id || '').snakeCase(xmlRe);
```

Comment:
@orangejenny I don't know why but this schedule doesn't work on the master branch too, I tried to verify where is the problem and I found this, when we create a new form with the visit schedule then ```schedule_form_id``` in the backend is None which give us in js undefined. So when we try to change the value of this property to snake case then we get an error ```Cannot read property 'replace' of null```.

 I didn't find why this occurred now. More context in case: https://manage.dimagi.com/default.asp?288614

Reply:
Thank you for digging into this - turns out this was reported back in 2017 but hadn't been prioritized to fix: http://manage.dimagi.com/default.asp?258379

Reply:
@orangejenny Ok, so we can check this functionality on staging and if all will work fine we can merge with refactoring or I should send this in other PR?

Reply:
With this PR is fine.